src/Microsoft.AspNet.Mvc.Core/ModelBinding/TypeMatchModelBinder.cs

```diff
@@ -19,7 +19,14 @@ public sealed class TypeMatchModelBinder : IModelBinder
             bindingContext.ModelState.SetModelValue(bindingContext.ModelName, valueProviderResult);
             var model = valueProviderResult.RawValue;
             ModelBindingHelper.ReplaceEmptyStringWithNull(bindingContext.ModelMetadata, ref model);
-            return new ModelBindingResult(model, bindingContext.ModelName, isModelSet: true);
+            return new ModelBindingResult(
+                model,
+                bindingContext.ModelName,
+                isModelSet: true,
+                validationNode: new ModelValidationNode(
+                    bindingContext.ModelName,
+                    bindingContext.ModelMetadata,
+                    model));
```

Comment:
Looks like increased complexity of the simplest model binder is a fallout of this change. Users extending this system to bind individual properties (leaf nodes) should not need to understand `ModelValidationNode` _too_.  Note the concept count here is a regression compared to MVC 6 before `ModelValidationNode` was removed.